<?xml version="1.0"?>
<launch>
  <arg name="robot_name" default="$(env ROBOT_NAME)"/>

  <!-- Merge front and back laser into topic "/scan" -->
  <node pkg="ira_laser_tools" name="laserscan_multi_merger" type="laserscan_multi_merger">
    <param name="destination_frame" value="$(arg robot_name)/base_link"/>
    <param name="cloud_destination_topic" value="/merged_cloud"/>
    <param name="scan_destination_topic" value="/scan"/>
    <param name="laserscan_topics"
           value ="/$(arg robot_name)/front_laser /$(arg robot_name)/back_laser" />
  </node>

  <!-- Reconfigure min and max angle of result "/scan" topic -->
  <node name="reconfigure_min_ang" pkg="dynamic_reconfigure" type="dynparam" args="set /laserscan_multi_merger angle_min -3.14"/>
  <node name="reconfigure_max_ang" pkg="dynamic_reconfigure" type="dynparam" args="set /laserscan_multi_merger angle_max 3.14"/>

  <node name="obstacle_extractor" pkg="obstacle_detector" type="obstacle_extractor_node">
    <param name="active"               value="true"/>
    <param name="use_scan"             value="true"/>
    <param name="use_pcl"              value="false"/>

    <param name="use_split_and_merge"    value="true"/>
    <param name="circles_from_visibles"  value="true"/>
    <param name="discard_converted_segments" value="true"/>
    <param name="transform_coordinates"  value="true"/>

    <param name="min_group_points"     value="5"/>

    <param name="max_group_distance"   value="0.5"/>
    <param name="distance_proportion"  value="0.00628"/>
    <param name="max_split_distance"   value="0.2"/>
    <param name="max_merge_separation" value="0.2"/>
    <param name="max_merge_spread"     value="0.2"/>
    <param name="max_circle_radius"    value="0.8"/>
    <param name="radius_enlargement"   value="0.3"/>

    <!-- <remap from="/scan" to="/bravo/scan"/> -->

    <param name="frame_id"             value="$(arg robot_name)/map"/>
  </node>

  <node name="obstacle_tracker" pkg="obstacle_detector" type="obstacle_tracker_node">
    <param name="active"                  value="true"/>

    <param name="loop_rate"               value="100.0"/>
    <param name="tracking_duration"       value="2.0"/>
    <param name="min_correspondence_cost" value="0.6"/>
    <param name="std_correspondence_dev"  value="0.15"/>
    <param name="process_variance"        value="0.1"/>
    <param name="process_rate_variance"   value="0.1"/>
    <param name="measurement_variance"    value="1.0"/>

    <param name="frame_id"                value="$(arg robot_name)/map"/>

    <remap from="tracked_obstacles" to="obstacles"/>
  </node>

  <!-- <node name="obstacle_publisher" pkg="obstacle_detector" type="obstacle_publisher_node">
    <param name="active"           value="false"/>
    <param name="reset"            value="false"/>

    <param name="fusion_example"   value="false"/>
    <param name="fission_example"  value="false"/>

    <param name="loop_rate"        value="10.0"/>
    <param name="radius_margin"    value="0.25"/>

    <param name="frame_id"         value="/map"/>
  </node> -->

</launch>
